src/analysis/alt_series.lean

```diff
@@ -0,0 +1,192 @@
+        apply add_le_of_nonpos_of_le _ (hk k (le_succ k)),
+        simp only [add_zero, neg_add_le_iff_le_add],
+        exact ha_anti (lt_add_one k).le, } },
+    { rw [neg_one_pow_of_odd hodd, neg_one_mul, add_neg_le_iff_le_add'],
+      apply le_trans (hk k rfl.ge),
+      exact le_add_of_nonneg_left (ha_nonneg k), } },
+end
+
+lemma alternating_partial_sum_diff_le_first
+  {m n : ℕ}
+  (ha_anti : antitone a)
+  (ha_nonneg : ∀ n, 0 ≤ a n)
+  (hmn : m ≤ n)
+  (hm : even m) :
+  alternating_partial_sum a n - alternating_partial_sum a m ≤ a m :=
+begin
+  unfold alternating_partial_sum,
+  simp only [← sum_Ico_eq_sub _ hmn, sum_Ico_eq_sum_range, hm.neg_one_pow_add_left],
+  apply alternating_partial_sum_le_first _ (anti_suffix_is_anti a ha_anti),
+  exact λ N, ha_nonneg (m + N),
+end
+
+-- Used to prove `cauchy_seq` for both `a n` and `-(a n)`
+@[nolint ge_or_gt] -- see Note [nolint_ge]
+lemma alternating_partial_sum_is_cau_seq
+  (ha_tendsto_zero : tendsto a at_top (𝓝 0))
+  (ha_anti : antitone a)
+  (ha_nonneg : ∀ (n : ℕ), 0 ≤ a n) :
+  ∀ (ε : ℝ),
+    ε > 0 →
+    (∃ (N : ℕ),
+       ∀ (n : ℕ),
+         n ≥ N → |alternating_partial_sum a n - alternating_partial_sum a N| < ε) :=
+begin
+  intros ε hε,
+
+  -- Convert `tendsto` to `ε`-based definition of limit
+  simp_rw [metric.tendsto_at_top, real.dist_0_eq_abs, abs_of_nonneg (ha_nonneg _)]
+    at ha_tendsto_zero,
+  obtain ⟨m, ha_tendsto_zero⟩ := ha_tendsto_zero ε hε,
+
+  have : ∀ (m : ℕ) (hatz : ∀ (n : ℕ), n ≥ m → a n < ε) (hme : even m),
+    ∀ (n : ℕ), n ≥ m → |alternating_partial_sum a n - alternating_partial_sum a m| < ε,
+  { intros m hatz hme n hmn,
+    rw abs_of_nonneg (alternating_partial_sum_diff_nonneg a ha_anti ha_nonneg hmn hme),
+    refine lt_of_le_of_lt _ (hatz m rfl.ge),
+    exact alternating_partial_sum_diff_le_first a ha_anti ha_nonneg hmn hme },
+
+  obtain hme | hmo := even_or_odd m,
+  { exact ⟨m, this m ha_tendsto_zero hme⟩, },
+  { use m+1,
+    apply this,
+    { intros n hn,
+      exact ha_tendsto_zero _ (le_of_succ_le hn), },
+    { simpa with parity_simps using hmo, } },
+end
+
+theorem alternating_partial_sum_is_cauchy_seq
+  (ha_tendsto_zero : tendsto a at_top (𝓝 0))
+  (ha_anti : antitone a)
+  (ha_nonneg : ∀ n, 0 ≤ a n) :
+  cauchy_seq (alternating_partial_sum a) :=
+begin
+  simp only [metric.cauchy_seq_iff', real.dist_eq],
+  exact alternating_partial_sum_is_cau_seq a ha_tendsto_zero ha_anti ha_nonneg,
+end
+
+theorem alternating_partial_sum_is_cauchy_seq'
+  (ha_tendsto_zero : tendsto a at_top (𝓝 0))
+  (ha_anti : antitone a)
+  (ha_nonneg : ∀ n, 0 ≤ a n) :
+  cauchy_seq (alternating_partial_sum' a) :=
+begin
+  simp only [metric.cauchy_seq_iff', alternating_partial_sum', real.dist_eq,
+    neg_sub_neg, sum_neg_distrib, mul_neg_eq_neg_mul_symm],
+  have hcs := alternating_partial_sum_is_cau_seq a ha_tendsto_zero ha_anti ha_nonneg,
+  simpa only [abs_sub_comm] using hcs,
```

Comment:
I forgot last night that you can give an expression to `simpa`:
```suggestion
  simpa only [abs_sub_comm]
    using alternating_partial_sum_is_cau_seq a ha_tendsto_zero ha_anti ha_nonneg,
```

Another thing you can do is
```suggestion
  have := alternating_partial_sum_is_cau_seq a ha_tendsto_zero ha_anti ha_nonneg,
  simpa only [abs_sub_comm],
```
since `simpa` defaults to looking at `this`.